internal/oci/container.go

```diff
@@ -72,6 +72,9 @@ type Container struct {
 	stdinOnce          bool
 	created            bool
 	spoofed            bool
+	stopping           bool
+	stopTimeoutChan    chan time.Duration
```

Comment:
I think we might get rid of `stopping` field by initializing `stopTimeoutChan` in `waitContainerStop`, and checking if it's nil in `StopContainer` (i.e. the idea is non-nil `stopTimeoutChan` means we're stopping). The only thing is to make sure to close stopTimeoutChan and set it to nil when exiting `waitContainerStop`.

Seems that `stopLock` is still required though.

Reply:
I'm not sure this will result in better code though -- just an idea, feel free to ignore.

Reply:
we can't close stopTimeoutChan upon leaving waitContainerStop, as we race against other routines putting a value in (risking inserting to a closed chan and panicing). I can't think of a way to avoid the bool :\